java/src/com/google/template/soy/parsepasses/contextautoesc/Rewriter.java

```diff
@@ -102,7 +103,11 @@
      * Keep track of template nodes so we know which are derived and which aren't.
      */
     @Override protected void visitTemplateNode(TemplateNode templateNode) {
-      Preconditions.checkState(!visitedTemplateNames.contains(templateNode.getTemplateName()));
+      if (visitedTemplateNames.contains(templateNode.getTemplateName())) {
+        throw SoySyntaxExceptionUtils.createWithNode(
+            "was defined previously",
+            templateNode);
+      }
```

Comment:
Thanks @mikesamuel, I went to implement your suggestion, and then discovered that the [TemplateRegistry](https://github.com/google/closure-templates/blob/master/java/src/com/google/template/soy/soytree/TemplateRegistry.java) was already changed to catch duplicate template definitions with a nice error message.

Can probably just close this pull request as no longer necessary then.